server/channels/api4/scheduled_post.go

```diff
@@ -47,6 +47,7 @@ func createSchedulePost(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 	scheduledPost.UserId = c.AppContext.Session().UserId
+	scheduledPost.CreateAt = 0
```

Comment:
While this works, is not a very robust nor cleanest solution and I'd say is a bad practice we all have made in the past. When handling JSON input it's better practice to explicitly define (via an input struct) which fields one might want to accept from the request body, instead of than decoding directly into the desired model struct. My suggestion here would be:
1) Define an input struct:
```
type ScheduledPostInput struct {
    // Include here only the fields you want to accept that where sent from the client
   ...
}
```
2) decode the request body into the input struct:
```
var input ScheduledPostInput
if err := json.NewDecoder(r.Body).Decode(&input); err != nil {
```

3) Map the input fields into the model struct (i.e):
```
scheduledPost := &model.ScheduledPost{
    ChannelId: input.ChannelId,
```
With this approach you will only be accepting specific fields you want and you prevent clients from manipulating fields they shouldn't have control on. Also, the client should be fixed to not send the create_at but server must be robust enough to handle this.

Thoughts @harshilsharma63 @esarafianou ?

Reply:
+1 on your suggestion

Reply:
Thats just not something we do in Mattermost server codebase, so not doing it here. Not that this won't work, but its just not something we do, so keeping the code consistent and keeping it this way.